validators/checks.py

```diff
@@ -885,17 +885,19 @@ def compare_to_solution(self, solution: str, translator: Translator, **kwargs) -
         """Compare the submission to the solution html."""
 
         def _inner(_: BeautifulSoup):
-            from validators.structure_validator import compare
+            from validators.structure_validator import compare, get_similarity
             from exceptions.structure_exceptions import NotTheSame
             try:
                 compare(solution, self.content, translator, **kwargs)
             except NotTheSame as err:
-                with Message(err.message_str()):
+                res = get_similarity(solution, self.content)
+                html_sim_str = f"\n Html {translator.translate(Translator.Text.SIMILARITY)}: {round(res[0] * 100)}%"
+                css_sim_str = f"\n Css {translator.translate(Translator.Text.SIMILARITY)}: {round(res[1] * 100)}%" if res[1] != 1 else ""
```

Comment:
Slightly more readable way of doing this by unpacking the tuple
```suggestion
                html_sim, css_sim = get_similarity(solution, self.content)
                html_sim_str = f"\n Html {translator.translate(Translator.Text.SIMILARITY)}: {round(html_sim * 100)}%"
                css_sim_str = f"\n Css {translator.translate(Translator.Text.SIMILARITY)}: {round(css_sim* 100)}%" if css_sim != 1 else ""
```